File: SDKs/amazonaws.com/lightsail/2016-11-28/go/pkg/models/operations/getalarms.go
package operations

import (
	"openapi/pkg/models/shared"
)

type GetAlarmsXAmzTargetEnum string

const (
	GetAlarmsXAmzTargetEnumLightsail20161128GetAlarms GetAlarmsXAmzTargetEnum = "Lightsail_20161128.GetAlarms"
)

type GetAlarmsHeaders struct {
	XAmzAlgorithm     *string                 `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                 `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                 `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                 `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                 `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                 `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                 `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        GetAlarmsXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type GetAlarmsRequest struct {
	Headers GetAlarmsHeaders
	Request shared.GetAlarmsRequest `request:"mediaType=application/json"`
}

type GetAlarmsResponse struct {
	AccessDeniedException     *interface{}
	ContentType               string
	GetAlarmsResult           *shared.GetAlarmsResult
	InvalidInputException     *interface{}
	NotFoundException         *interface{}
	OperationFailureException *interface{}
	ServiceException          *interface{}
	StatusCode                int64
	UnauthenticatedException  *interface{}
}

File: SDKs/amazonaws.com/lightsail/2016-11-28/typescript/src/sdk/models/operations/getalarms.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";

export enum GetAlarmsXAmzTargetEnum {
    Lightsail20161128GetAlarms = "Lightsail_20161128.GetAlarms"
}


export class GetAlarmsHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: GetAlarmsXAmzTargetEnum;
}


export class GetAlarmsRequest extends SpeakeasyBase {
  @Metadata()
  headers: GetAlarmsHeaders;

  @Metadata({ data: "request, media_type=application/json" })
  request: shared.GetAlarmsRequest;
}


export class GetAlarmsResponse extends SpeakeasyBase {
  @Metadata()
  accessDeniedException?: any;

  @Metadata()
  contentType: string;

  @Metadata()
  getAlarmsResult?: shared.GetAlarmsResult;

  @Metadata()
  invalidInputException?: any;

  @Metadata()
  notFoundException?: any;

  @Metadata()
  operationFailureException?: any;

  @Metadata()
  serviceException?: any;

  @Metadata()
  statusCode: number;

  @Metadata()
  unauthenticatedException?: any;
}
